django-rgd/rgd/rest/viewsets.py

```diff
@@ -18,6 +19,24 @@ class CollectionViewSet(ModelViewSet):
     queryset = models.Collection.objects.all()
     filterset_class = CollectionFilter
 
+    @swagger_auto_schema(
+        method='GET',
+        operation_summary='Get the file at the given index in this collection. Associated files are orderd by primary key and the provided index is the index in that ordered set. If the files in the collection change, this will not be reproducible.',
+    )
+    @action(
+        detail=True,
+        serializer_class=serializers.ChecksumFileSerializer,
+        url_path=r'item/(?P<index>\d+)',
+    )
+    def item(self, request, pk, index):
+        collection = models.Collection.objects.get(pk=pk)
```

Comment:
```suggestion
        collection = get_object_or_404(models.Collection, pk=pk)
```
`Collection.objects.get` raises an exception if the object doesn't exist, so this will return a 500 error if trying to get a Collection that doesn't exist.